src/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -298,6 +298,37 @@ static void EnsureCapacity(char[]? dest, int destSize, int requiredSize)
             }
         }
 
+        internal static unsafe char[] UnescapeString(string input, int start, int end, char[] dest,
+            ref int destPosition, char rsvd1, char rsvd2, char rsvd3, UnescapeMode unescapeMode, UriParser? syntax,
+            bool isQuery)
+        {
+            fixed (char* pStr = input)
+            {
+                return UnescapeString(pStr, start, end, dest, ref destPosition, rsvd1, rsvd2, rsvd3, unescapeMode,
+                    syntax, isQuery);
+            }
+        }
+
+        internal static unsafe char[] UnescapeString(char* pStr, int start, int end, char[] dest, ref int destPosition,
+            char rsvd1, char rsvd2, char rsvd3, UnescapeMode unescapeMode, UriParser? syntax, bool isQuery)
+        {
+            ValueStringBuilder pooledArray = new ValueStringBuilder(dest.Length);
+            pooledArray.Append(dest.AsSpan(0, destPosition));
+
+            UnescapeString(pStr, start, end, ref pooledArray, rsvd1, rsvd2, rsvd3, unescapeMode,
+                    syntax, isQuery);
+            destPosition = pooledArray.Length;
+
+            if (destPosition > dest.Length)
+            {
+                dest = new char[destPosition];
+            }
+
+            pooledArray.AsSpan().TryCopyTo(dest);
```

Comment:
What does `TryCopyTo` do? And I assume it returns a bool true/false? Why is it ok to ignore that return value?

Reply:
[It copies the current span content to the target](https://github.com/dotnet/corefx/blob/7cae92b39d2fed2e4e88e900f2e7d787ed9f6cfa/src/Common/src/CoreLib/System/ReadOnlySpan.Fast.cs#L194). Method returns false only when the target's length is shorter that the source's one. Here, before copying, I ensure that "dest" has enough length to hold all the data, so the call will always succeed.